Module2/problem-1/problem1.java

```diff
@@ -0,0 +1,161 @@
+import java.io.IOException;
+import java.util.Comparator;
+import java.util.Map;
+import java.util.LinkedHashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Scanner;
+import java.util.TreeMap;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+
+
+
+class WebContent {
+
+    // getting the text from website using the url
+    String parseUrl(String url) {
+        String content = "";
+        try {
+            // Get the complete document
+            Document document = Jsoup.connect(url).get();
+            // Get all the text from text tags
+            content = document.body().text();
+            // Regular Expression to remove any special character or unwanted spaces
+            content = content.replaceAll("[^a-zA-Z\\w]", " ");
+        } catch (IOException e) {
+            System.out.println(e);
+        }
+        // return the content
+        return content;
+    }
+
+    // Function to count the frequency of the given word
+    int countWordOccurence(String word, String urlContent) {
+        int wordCount = 0;
+        // Using the .split method to remove space and store the content into array of string
+        String contentList[] = urlContent.split(" ");
+        for (String str : contentList) {
+            // find the word in the text
+            if (str.equalsIgnoreCase(word)) {
+                // if the word is found then increment the counter for words
+                wordCount++;
+            }
+        }
+        //return the number of occurences of the word
+        return wordCount;
+    }
+
+    // Sort the Map by Value
+    public static Map<String, Integer> sortByValue(Map<String, Integer> hm) {
+        // Create a list from elements of HashMap
+        List<Map.Entry<String, Integer>> lists = new LinkedList<Map.Entry<String, Integer>>(
+                hm.entrySet());
+
+        // Sort the lists
+        Collections.sort(
+                lists,
+                new Comparator<Map.Entry<String, Integer>>() {
+                    public int compare(
+                            Map.Entry<String, Integer> o1,
+                            Map.Entry<String, Integer> o2) {
+                        return (o2.getValue()).compareTo(o1.getValue());
+                    }
+                });
+
+        // put data from sorted list to hashmap
+        Map<String, Integer> temp = new LinkedHashMap<String, Integer>();
+        for (Map.Entry<String, Integer> aa : lists) {
+            temp.put(aa.getKey(), aa.getValue());
+        }
+        return temp;
+    }
+
+}
+
+// Driver / main Class
+
+public class Main {
+
+    public static void main(String[] args) {
+        try {
+            // Get URLS from the text file
+
+            ArrayList<String> urls = new ArrayList<>();
+            Scanner Reader = new Scanner(new File("//home//rishabh//Desktop//url.txt"));
+
+            while (Reader.hasNext()) {
```

Comment:
Handle all the edgeCases properly eg. if urls or words files are empty , or missing files , or Invalid URL given etc.

Reply:
resolved